src/utils/index.tsx

```diff
@@ -209,8 +209,24 @@ export function createNotFoundResponse() {
   return createPreactResponse(<NotFound />, { status: 404 })
 }
 
-async function toCacheKey(key: string): Promise<string> {
-  return key.length > 512 ? await digestMessage(key) : key
+interface CacheKeyBrand {
+  readonly CacheKey: unique symbol
+}
+
+const CacheKey = t.brand(
+  t.string,
+  (text): text is t.Branded<string, CacheKeyBrand> => text.length <= 512,
+  'CacheKey'
+)
+
+export type CacheKey = t.TypeOf<typeof CacheKey>
+
+export async function toCacheKey(key: string): Promise<CacheKey> {
+  const shortenKey = key.length > 512 ? await digestMessage(key) : key
+
+  return E.getOrElse<unknown, CacheKey>(() => {
+    throw new Error('Illegal State')
+  })(CacheKey.decode(shortenKey))
```

Comment:
Can you explain to me why this gives back the key in the good case and not always the Error?

Reply:
`CacheKey.decode(shortenKey)` gibt `Either` zurück`. Nun ist `getOrElse` folgendermaßen geschrieben:

```ts
function getOrElse(whenLeftFunc) {
  return (either) => {
    if (isLeft(either)) {
      return whenLeftFunc(either.left)
    } else {
      return either.right
    }
  }
}
```

Die Funktion wird also nur im error-Fall ausgeführt (beachte, dass die throw-Anweisung innerhalb einer ArrowFunktion ist und nicht direkt ausgeführt wird)

Reply:
Thank you!